azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -258,6 +266,41 @@ func resourceArmKubernetesCluster() *schema.Resource {
 				Set: resourceAzureRMKubernetesClusterServicePrincipalProfileHash,
 			},
 
+			"aad_profile": {
+				Type:     schema.TypeSet,
```

Comment:
after some experimentation/research here I think this'd be better within Terraform as:

```
role_based_access_control {
  azure_active_directory {
    client_app_id = "..."
    server_app_id = "..."
    sertver_app_secret = "..."
    tenant_id = "..." # it also appears this could be optional + defaulted from the provider block
  }
}
```

again after some experimentation here I believe we should be able to make this a List rather than a Set - which would allow us to determine when each of the fields has changed. whilst that means we'd need to delve into the state to obtain the Client Secret (since it's not returned from the API, and thus won't be available during Import) - it provides a more consistent approach across Terraform where a block being absent indicates that the thing shouldn't be enabled/active.

Reply:
(removed in the newly pushed commits)